FWCore/Framework/test/stubs/TestStreamProducers.cc

```diff
@@ -198,19 +181,17 @@ namespace edmtest {
                                                                edm::EventSetup const&,
                                                                RunContext const*) {
         ++m_count;
-        gbl = true;
-        gel = false;
         auto pCache = std::make_shared<Cache>();
-        ++(pCache->lumi);
+        pCache->run = iLB.luminosityBlockAuxiliary().run();
+        pCache->lumi = iLB.luminosityBlockAuxiliary().luminosityBlock();
         return pCache;
       }
 
-      void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override {
-        bl = true;
-        el = false;
-        if (!gbl) {
+      void beginLuminosityBlock(edm::LuminosityBlock const& iLB, edm::EventSetup const&) override {
+        if (luminosityBlockCache()->run != iLB.luminosityBlockAuxiliary().run()) {
```

Comment:
Should this check also the LuminosityBlock number?

Reply:
Done.